pyproject.toml

```
@@ -0,0 +1,80 @@
+[build-system]
+requires = ["flit_core >=3.2,<4"]
+build-backend = "flit_core.buildapi"
+
+[project]
+name = "py2spack"
+authors = [{name = "davhofer", email = "davhofer@ethz.ch"}]
+readme = "README.md"
+license = {text = "MIT"}
+classifiers = ["License :: OSI Approved :: MIT License"]
+dynamic = ["version", "description"]
+dependencies = [
+    "packaging>=20", 
+    "requests>=2.30", 
+    "tomli>=1.2",
+]
+
+[project.urls]
+Home = "https://github.com/davhofer/py2spack/"
+
+
+[tool.ruff]
+    exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+]
+
+line-length = 80
+indent-width = 4
+
+[tool.ruff.lint]
+select = ["E", "F", "W", "B", "N", "D"]
+ignore = []
+
+
+[tool.ruff.lint.pydocstyle]
+convention = "google"
+
+[tool.ruff.format]
+skip-magic-trailing-commas = false 
+
+
+[tool.mypy]
+warn_return_any = true  
+warn_unused_configs = true
+ignore_missing_imports = true
+check_untyped_defs = true
+warn_unreachable = true 
+
+[tool.pytest.ini_options]
+minversion = "6.0"
+addopts = "-ra -q"
```

I'd also suggest to use more strict settings here.
```suggestion
addopts = ["-ra", "--showlocals", "--strict-markers", "--strict-config"]
```